experiments/dorop-20/metadata/element_adapter.py

```diff
@@ -0,0 +1,46 @@
+import xml.etree.ElementTree as ET
+from xml.etree.ElementTree import Element
+
+from metadata.exceptions import DataNotFoundError
+
+class ElementAdapter():
+
+    @classmethod
+    def from_string(cls, text: str, namespaces: dict[str, str]) -> "ElementAdapter":
+        return cls(ET.fromstring(text=text), namespaces)
+
+    def __init__(self, elem: Element, namespaces: dict[str, str]):
+        self.elem = elem
+        self.namespaces = namespaces
+
+    def find(self, path: str) -> "ElementAdapter":
+        result = self.elem.find(path, self.namespaces)
+        if result is None:
+            raise DataNotFoundError()
+        return ElementAdapter(result, self.namespaces)
```

Comment:
I did end up implementing a thin wrapper around the standard library `xml.etree.ElementTree.Element` so that I could raise errors when things were found to be missing. I could imagine this being modified slightly to accommodate `None` as a return value in some cases (maybe by adding a `required=True` parameter to `find`, `get`, and `text`).